### PR TITLE
Update ui-class.md

### PR DESCRIPTION
**Pull request by fiyazbinhasan**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/20788

@Rick-Anderson, please review

### CODE REVIEW

File: aspnetcore/razor-pages/ui-class/samples/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+```csharp
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/ui-class/samples/Startup.cs:1]: Make a complete startup file

File: aspnetcore/razor-pages/ui-class.md [hunk 2]
```diff
@@ -446,6 +446,33 @@ In the sample download, rename *WebApp1/Areas/MyFeature2* to *WebApp1/Areas/MyFe
 
 Copy the *RazorUIClassLib/Areas/MyFeature/Pages/Shared/_Message.cshtml* partial view to *WebApp1/Areas/MyFeature/Pages/Shared/_Message.cshtml*. Update the markup to indicate the new location. Build and run the app to verify the app's version of the partial is being used.
 
+One must enable `Razor Pages` in the web app to make sure the page based routing works. For example, in an MVC application, register the `Razor Pages` services and map the pages with endpoint routing,
+
+```csharp
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/ui-class.md:451]: Finish moving this to file. Use snippet syntax to highlight the two lies
Example syntax
`[!code-csharp[](start-mvc/sample/src/MvcMovie/Models/MovieDateFormat.cs?name=snippet1&highlight=2,10)]`

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/razor-pages/ui-class.md [hunk 3]
```diff
@@ -446,6 +446,10 @@ In the sample download, rename *WebApp1/Areas/MyFeature2* to *WebApp1/Areas/MyFe
 
 Copy the *RazorUIClassLib/Areas/MyFeature/Pages/Shared/_Message.cshtml* partial view to *WebApp1/Areas/MyFeature/Pages/Shared/_Message.cshtml*. Update the markup to indicate the new location. Build and run the app to verify the app's version of the partial is being used.
 
+One must enable `Razor Pages` in the web app to make sure the page based routing works. For example, in an MVC application, register the `Razor Pages` services and map the pages with endpoint routing,
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/ui-class.md:449]:
```suggestion
If Razor Pages are used in the RCL, Razor Pages services and endpoints must be enabled in the hosting app:
```

**Comment by Rick-Anderson** [comment]
@serpent5  please review

File: aspnetcore/razor-pages/ui-class.md [hunk 4]
```diff
@@ -446,6 +446,10 @@ In the sample download, rename *WebApp1/Areas/MyFeature2* to *WebApp1/Areas/MyFe
 
 Copy the *RazorUIClassLib/Areas/MyFeature/Pages/Shared/_Message.cshtml* partial view to *WebApp1/Areas/MyFeature/Pages/Shared/_Message.cshtml*. Update the markup to indicate the new location. Build and run the app to verify the app's version of the partial is being used.
 
+If Razor Pages are used in the RCL, Razor Pages services and endpoints must be enabled in the hosting app:
```
Comment (serpent5) [inline aspnetcore/razor-pages/ui-class.md:449]:
```suggestion
If the RCL uses Razor Pages, enable the Razor Pages services and endpoints in the hosting app:
```

**Review by serpent5 — APPROVED** [review]
Just a minor suggestion to switch from a passive to an active voice.

Nice work @fiyazbinhasan. ⭐

**Review by Rick-Anderson — APPROVED** [review]

**Comment by fiyazbinhasan** [comment]
thanks @serpent5 😃

**Comment by chengyanbing** [comment]
When I build it, it shows: ...aspnetcore\razor-pages\ui-class\samples\cli\WebApp1\Pages\Error.cshtml.cs(17,10,17,92): warning MVC1001: 'ResponseCacheAttribute' cannot be applied to Razor Page handler methods. It may be applied either to the Razor Page model or applied globally.

**Comment by fiyazbinhasan** [comment]
Nice catch. I'll add a fix for that. @chengyanbing

**Pull request merged**